src/mlpack/tests/augmented_rnns_tasks_test.cpp

```diff
@@ -95,7 +95,8 @@ class HardCodedSortModel
   void Train(arma::field<arma::mat>& predictors,
              arma::field<arma::mat>& labels)
   {
-    assert(predictors.n_elem == labels.n_elem);
+    const bool check = predictors.n_elem == labels.n_elem;
+    assert(check == true);
```

Comment:
Interesting, I guess this one is a little tricker---I wonder if just using `Log::Assert()` could fix the unused variable warning.